app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_source_form.xml

```diff
@@ -54,6 +54,7 @@
             <settings>
                 <validation>
                     <rule name="required-entry" xsi:type="boolean">true</rule>
+                    <rule name="no-whitespace" xsi:type="boolean">true</rule>
```

Comment:
Also, need to add check to `\Magento\Inventory\Model\Source\Validator\CodeValidator`
And expand API tests

Reply:
[app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php](https://github.com/magento-engcom/msi/blob/develop/app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php)
This class contains already this check.


```
elseif (preg_match('/\s/', $value)) {
            $errors[] = __('"%field" can not contain whitespaces.', ['field' => SourceInterface::SOURCE_CODE]);
        }
```

[Magento/Inventory/Test/Unit/Model/Source/Validator/CodeValidatorTest.php](https://github.com/magento-engcom/msi/blob/develop/app/code/Magento/Inventory/Test/Unit/Model/Source/Validator/CodeValidatorTest.php)
And it is also covered by test.

```
public function testValidateCodeNotWithWhiteSpaces()
    {
        $emptyValidatorResult = $this->createMock(\Magento\Framework\Validation\ValidationResult::class);
        $this->validationResultFactory->expects($this->once())
            ->method('create')
            ->with([
                'errors' => [__('"%field" can not contain whitespaces.', ['field' => SourceInterface::SOURCE_CODE])]
            ])
            ->willReturn($emptyValidatorResult);
        $this->codeValidator = (new ObjectManager($this))->getObject(CodeValidator::class, [
            'validationResultFactory' => $this->validationResultFactory
        ]);
        $this->source->expects($this->once())
            ->method('getSourceCode')
            ->willReturn(' source code ');
        $this->codeValidator->validate($this->source);
    }
```